snap/info.go

```diff
@@ -196,6 +196,7 @@ type SideInfo struct {
 	EditedDescription string   `yaml:"description,omitempty" json:"description,omitempty"`
 	Private           bool     `yaml:"private,omitempty" json:"private,omitempty"`
 	Paid              bool     `yaml:"paid,omitempty" json:"paid,omitempty"`
+	DisabledServices  []string `yaml:"disabled-services,omitempty" json:"disabled-services,omitempty"`
```

Comment:
Keeping this is SideInfo feels wrong as it's part of the local state. Then SnapState is not a good match either. It does keep the state (eg. instance-key), but services can be easily disabled outside of snapd, so we'd either have to keep this state in sync with systemd or fallback to asking systemd anyway. Since it's really an ephemeral piece of information only useful on install/revert/refresh. I feel this should actually be a part of SnapSetup and passed around. The refresh/revert code can build a list of disabled services at the right time.

Reply:
I didn't think much about the whole change but I think I agree with Maciej on the key aspect: it's not really our state. We provide a means for the snap to change it but there are other means of changing that state that do not interact with snapd (systemd presets, systemd state). I agree that we should really only keep this around inside a task/change state, not anywhere else.

Reply:
as I explained in the PR description:

> This is more like a snapshot of the service state at a particular point in time than it is a current service state anyways, so code operating on active snaps shouldn't be looking at this anyways, they should be querying systemd to get the current state.

This _is_ our state because we need this information in order to manage services - we expose service management options to users with things like `snap stop --disable` that currently is lost on refresh/revert/disable. So when we refresh, revert, or disable a snap, snapd needs to save a snapshot of those service states at the particular point in time that snapd is modifying the services. Later in the same task we disable and then remove the systemd service which erases the user's systemd state, so this state is really only meant to be used by snapd during the time we are switching snap revisions. It could be ephemeral and not persistent inside any snap* structures, but that wouldn't then solve the problem with revert/disable which could happen at any time later after a revision is unlinked.

Reply:
what other have remarked here is correct, see my other comment about a similar case of cross-task but change only state

Reply:
I discussed this over IRC and went with a different approach. Please re-read the PR description